### ISSUE TITLE
Is stop_trace_locked needed?

### ISSUE THREAD

**Issue by edwardalee** [lf-lang/reactor-c#314]
@erlingrj added a `stop_trace_locked` in `trace.c` and `trace.h`, but I'm not sure why. The only place it is called seems to be in the `termination` function, and it seems to be called without holding any locks.  Maybe the reason is that acquiring locks during termination is risky because it can prevent a program from exiting that has deadlocked on these same locks.  Was this the reason for this change?  If so, I guess probably all we need is a comment in the `termination` function explaining why this is called instead of `stop_trace`.

Alternatively, we should just have `stop_trace` always acquire the lock?  But as I said, this is risky in the termination function.

**Comment by erlingrj** [comment]
You are correct. I experienced a deadlock in termination. For some reason, this hadn't occurred until multiple environments were introduced. Could it be that some other worker held this mutex when being terminated? I don't exactly recall this.
